File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/ConnectNodeAppendedImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result appending a node connection
 */
class ConnectNodeAppendedImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * all edges that are set after this request
     */
    fun allEdges(config: Edge.() -> Unit) = 
        add { "allEdges{ " + Edge().apply(config).render(it) + "}"}


    /**
     * edge created by this request
     */
    fun newEdge(config: Edge.() -> Unit) = 
        add { "newEdge{ " + Edge().apply(config).render(it) + "}"}

}

File: gaia-sdk-javascript/src/graphql/response/type/ConnectNodeAppendedImpulse.ts

import {Edge} from "./Edge";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result appending a node connection
*/
export interface ConnectNodeAppendedImpulse {
    id?:Uuid, 
    /**
    * edge created by this request
    */
    newEdge?:Edge
}